sources/osg/Program.js

```diff
@@ -82,6 +94,14 @@ Program.prototype = MACROUTILS.objectLibraryClass( MACROUTILS.objectInherit( GLO
         return new Program();
     },
 
+    dirty: function () {
```

Comment:
there is no rules, some resources needs to be re uploaded some not. A StateAttribute is not always a resources and it depends how it's used. So for Texture it makes sense but not for CullFace. I tried to generalize but it does not make sense. So let's say that it depends on context/usage